isthmus/src/main/java/io/substrait/isthmus/SqlToSubstrait.java

```diff
@@ -48,6 +62,26 @@ public Plan execute(String sql, String name, Schema schema) throws SqlParseExcep
     return executeInner(sql, factory, pair.left, pair.right);
   }
 
+  /**
+   * Process to execute an SQL Expression to convert into an Extended expression protobuf message
+   *
+   * @param sqlExpression expression defined by the user
+   * @param tables of names of table needed to consider to load into memory for catalog, schema,
+   *     validate and parse sql
+   * @return extended expression protobuf message
+   * @throws SqlParseException
+   */
+  public ExtendedExpression executeSQLExpression(String sqlExpression, List<String> tables)
+      throws SqlParseException {
+    var result = registerCreateTablesForExtendedExpression(tables);
+    return executeInnerSQLExpression(
+        sqlExpression,
+        result.validator(),
+        result.catalogReader(),
+        result.nameToTypeMap(),
+        result.nameToNodeMap());
+  }
```

Comment:
What do you think about moving all of the logic associated with converting SQL _expressions_ into a dedicated class like `SqlExpressionToSubstrait`? That way each class has one responsibility and folks looking to convert sql or sql expressions can just look at the specific class for the conversion they need to understand what's happening under the hood.

Reply:
Changed